src/Microsoft.ML.CpuMath/SseIntrinsics.cs

```diff
@@ -566,12 +566,12 @@ public static unsafe void Scale(float scale, Span<float> dst)
             }
         }
 
-        public static unsafe void ScaleSrcU(float scale, Span<float> src, Span<float> dst)
+        public static unsafe void ScaleSrcU(float scale, ReadOnlySpan<float> src, Span<float> dst, int count)
```

Comment:
>count [](start = 103, length = 5)

this addition also probably warrants a summary comment

Reply:
all of them I mean

---
In reply to: [224550873](https://github.com/dotnet/machinelearning/pull/1229#discussion_r224550873) [](ancestors = 224550873)

Reply:
Putting summary comments on SseIntrinsics is a waste because only CpuMathUtils will be calling SseIntrinsics. It is an internal class to `ML.CpuMath`. The summary comments should go on the external facting `CpuMathUtils` class.

I can add them to CpuMathUtils - do you want full summary comments for all functions?

Reply:
I suppose yes. Can we make these methods non-public then?

---
In reply to: [224636550](https://github.com/dotnet/machinelearning/pull/1229#discussion_r224636550) [](ancestors = 224636550)

Reply:
It's an internal class, so this whole class is internal.

Reply:
I've opened #1265 to add xml comments and arg validation since `CpuMathUtils` is public.